src/blockchaintree.rs

```diff
@@ -815,11 +820,12 @@ impl BlockChainTree {
         &mut self.main_chain
     }
 
-    pub fn create_derivative_chain(
+    pub async fn create_derivative_chain(
+        &mut self,
         addr: &[u8; 33],
         genesis_hash: &[u8; 32],
         global_height: u64,
-    ) -> Result<Box<DerivativeChain>, BlockChainTreeError> {
+    ) -> Result<&Arc<RwLock<DerivativeChain>>, BlockChainTreeError> {
```

Comment:
Result<Option<Arc<RwLock>>, BlockChainTreeError>

Reply:
![image](https://user-images.githubusercontent.com/87724011/212565785-392745c7-7a49-4c40-b217-d3d97a52e595.png)
thats also need <>

Reply:
![image](https://user-images.githubusercontent.com/87724011/212565840-d605c010-3bee-4435-a59b-4128d7d03d94.png)
Need <> and arguments

Reply:
> `Result<Option<Arc<RwLock<DerivativeChain>>>, BlockChainTreeError>`

fixed markdown